### PR TITLE
fix: Spawn points do not work in SDK7

### LINKED ISSUE #619: Spawn points do not work in SDK7
### **Explorer Alpha build version:**
Main

### **Issue Description:**
Spawn point in SDK7 is not respected.

### **Steps to reproduce:**

  1. Jump into DCLExperience.dcl.eth
  2. Notice you do not use the spawn point

### **Expected behaviour:**
It uses the existing spawn point.

### **Current behaviour:**
image

### PR DESCRIPTION
**Pull request by sandrade-dcl**
## What does this PR change?
Fix #619 

## How to test the changes?
1. Launch the explorer.
2. Check that the initial position of your avatar is correct.
3. Do different teleports.
4. Check that the new position of your avatar is correct.

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

File: Explorer/Assets/DCL/Character/CharacterMotion/Systems/InterpolateCharacterSystem.cs [hunk 1]
```diff
@@ -38,6 +39,9 @@ protected override void Update(float t)
         [Query]
         private void TeleportPlayer(in Entity entity, in CharacterController controller, ref CharacterPlatformComponent platformComponent, in PlayerTeleportIntent teleportIntent)
         {
+            if (teleportIntent.LoadReport != null && teleportIntent.LoadReport.CompletionSource.UnsafeGetStatus() == UniTaskStatus.Pending)
```
Comment (NickKhalow) [inline Explorer/Assets/DCL/Character/CharacterMotion/Systems/InterpolateCharacterSystem.cs:42]: Why is it checking by LoadReport? It's not marked as nullable

File: Explorer/Assets/Scripts/CrdtEcsBridge/RestrictedActions/GlobalWorldActions.cs [hunk 2]
```diff
@@ -21,7 +21,7 @@ public class GlobalWorldActions : IGlobalWorldActions
         public void MoveAndRotatePlayer(Vector3 newPlayerPosition, Vector3? newCameraTarget)
         {
             // Move player to new position (through InterpolateCharacterSystem -> TeleportPlayerQuery)
-            world.Add(playerEntity, new PlayerTeleportIntent(newPlayerPosition, Vector2Int.zero));
+            world.Add(playerEntity, new PlayerTeleportIntent(newPlayerPosition, Vector2Int.zero, null));
```
Comment (NickKhalow) [inline Explorer/Assets/Scripts/CrdtEcsBridge/RestrictedActions/GlobalWorldActions.cs:24]: nullables please

File: Explorer/Assets/Scripts/ECS/SceneLifeCycle/Systems/CheckCameraQualifiedForRepartitioningSystem.cs [hunk 3]
```diff
@@ -25,24 +26,41 @@ public partial class CheckCameraQualifiedForRepartitioningSystem : BaseUnityLoop
...
+            CheckCameraTransformChangedQuery(World);
         }
 
         [Query]
         private void CheckCameraTransformChanged(ref CameraSamplingData cameraSamplingData, ref CameraComponent cameraComponent)
         {
+            if (cameraSamplingData.IsDirty)
+                return;
+
             ScenesPartitioningUtils.CheckCameraTransformChanged(cameraSamplingData, in cameraComponent, partitionSettings.PositionSqrTolerance, partitionSettings.AngleTolerance);
```
Comment (NickKhalow) [inline Explorer/Assets/Scripts/ECS/SceneLifeCycle/Systems/CheckCameraQualifiedForRepartitioningSystem.cs:55]: PartitionDiscreteDataBase will be mutated in CheckCameraTransformChanged, please clarify the name with CQRS

**Review by NickKhalow — CHANGES_REQUESTED** [review]
Works correctly 💚
Please take a look on requests

File: Explorer/Assets/DCL/Character/CharacterMotion/Systems/InterpolateCharacterSystem.cs (hunk 1 above)
Comment (sandrade-dcl) [reply Explorer/Assets/DCL/Character/CharacterMotion/Systems/InterpolateCharacterSystem.cs:42]: LoadReport marked as nullable: https://github.com/decentraland/unity-explorer/pull/717/commits/76fce5394aa192c52e6e656a8ef7676df11a4437

File: Explorer/Assets/Scripts/CrdtEcsBridge/RestrictedActions/GlobalWorldActions.cs (hunk 2 above)
Comment (sandrade-dcl) [reply Explorer/Assets/Scripts/CrdtEcsBridge/RestrictedActions/GlobalWorldActions.cs]: Done: https://github.com/decentraland/unity-explorer/pull/717/commits/76fce5394aa192c52e6e656a8ef7676df11a4437

File: Explorer/Assets/Scripts/ECS/SceneLifeCycle/Systems/CheckCameraQualifiedForRepartitioningSystem.cs (hunk 3 above)
Comment (sandrade-dcl) [reply Explorer/Assets/Scripts/ECS/SceneLifeCycle/Systems/CheckCameraQualifiedForRepartitioningSystem.cs]: Done: https://github.com/decentraland/unity-explorer/pull/717/commits/7fd26b63d58f1d34237be46c1588f83b33b94794

File: Explorer/Assets/DCL/Character/CharacterMotion/Systems/InterpolateCharacterSystem.cs (hunk 1 above)
Comment (sandrade-dcl) [reply Explorer/Assets/DCL/Character/CharacterMotion/Systems/InterpolateCharacterSystem.cs:42]: Done: https://github.com/decentraland/unity-explorer/pull/717/commits/2ce7a9903a454fa406ae323a1dc6129ced10eb3e

**Review by NickKhalow — APPROVED** [review]

**Review by anicalbano — COMMENTED** [review]
🟢 Checked by QA

https://github.com/decentraland/unity-explorer/assets/139157886/5ade6b43-2d5c-4d34-a4cb-c7d31dec9e46

**Comment by LucasLioyQA** [comment]
**Initial position:** 
initial position ok

**Jumping into DCLExperience.dcl.eth world:** 
DCLExperience initial position ok

**Pull request merged**